doc/sphinxext/allow_nan.py

```diff
@@ -0,0 +1,98 @@
+from sklearn.utils import all_estimators
+from sklearn.linear_model import LogisticRegression
+from sklearn.preprocessing import OrdinalEncoder, OneHotEncoder
+from sklearn.compose import ColumnTransformer
+from io import StringIO
+from docutils import nodes
+
+from docutils.parsers.rst import Directive
+
+
+class Allow_Nan(Directive):
+    def run(self):
+        names1 = []
+        names2 = []
+        names3 = []
+        names4 = []
+        output = StringIO()
+        output.write(
+            f"List of estimators that allow NaN values for type_filter=cluster : "
+        )
+        for name, Estimator in all_estimators(type_filter="cluster"):
+            if Estimator()._get_tags().get("allow_nan") == True:
+                names1.append(name)
+        output.write(f" {names1} ")
+
+        output2 = StringIO()
+        output2.write(
+            f" List of estimators that allow NaN values for type_filter=regressor : "
+        )
+        for name, Estimator in all_estimators(type_filter="regressor"):
+            try:
+                if Estimator()._get_tags().get("allow_nan") == True:
+                    names2.append(name)
+            except TypeError:
+                base = LogisticRegression(solver="lbfgs")
+                if Estimator(base)._get_tags().get("allow_nan") == True:
+                    names2.append(name)
+        output2.write(f" {names2} ")
```

Comment:
It would be great to have a minimal styling and a link to the API. Something like
```
* :class:`module.FirstRegressor`
* :class:`module.SecondRegressor`
```

Reply:
Note that to construct the fully qualified name, you can combine the information found in `Estimator.__module__` and `name` (the later should match `Estimator.__name__`).